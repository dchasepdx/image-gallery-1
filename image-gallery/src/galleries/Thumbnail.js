import React from 'react';
import PropTypes from 'prop-types';

export default function Thumbnail(props) {


        return (
            <div>
                {props.images.map((image, index) => (
                    <figure key={index}
                        style={{
                        display: 'inline-block'
                    }}>
                        <img src={image.url} alt='' style={{
                            height: '100px',
                            width: '100px',
                        }}
                        />
                        <figcaption> {image.title} </figcaption>
                    </figure>
                ))}
            </div>

        )

}

Thumbnail.proptypes = {
images:PropTypes.arrayOf(PropTypes.object).isRequired
};  