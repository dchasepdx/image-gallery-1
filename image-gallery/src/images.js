import images from './imagesData';

export default {
    get() { return Promise.resolve(images) }
}