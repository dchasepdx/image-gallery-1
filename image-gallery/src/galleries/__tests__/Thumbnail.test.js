import React from 'react';
import ReactDOM from 'react-dom';
import Thumbnail from '../Thumbnail';
import renderer from 'react-test-renderer';
import images from '../../imagesData';



it('renders List view without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Thumbnail  images={images}/>, div);
});

it('renders a snapshot', () => {
  const tree = renderer.create(<Thumbnail images={images}/>).toJSON();
  expect(tree).toMatchSnapshot();
});