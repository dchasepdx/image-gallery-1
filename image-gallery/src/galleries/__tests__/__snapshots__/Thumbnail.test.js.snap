// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<div>
  <figure
    style={
      Object {
        "display": "inline-block",
      }
    }
  >
    <img
      alt=""
      src="http://f.cl.ly/items/3g3J1G0w122M360w380O/3726490195_f7cc75d377_o.jpg"
      style={
        Object {
          "height": "100px",
          "width": "100px",
        }
      }
    />
    <figcaption>
       
      Cute Bunny
       
    </figcaption>
  </figure>
  <figure
    style={
      Object {
        "display": "inline-block",
      }
    }
  >
    <img
      alt=""
      src="http://static.boredpanda.com/blog/wp-content/uploads/2015/09/cute-bunnies-25__605.jpg"
      style={
        Object {
          "height": "100px",
          "width": "100px",
        }
      }
    />
    <figcaption>
       
      Cuter Bunny
       
    </figcaption>
  </figure>
  <figure
    style={
      Object {
        "display": "inline-block",
      }
    }
  >
    <img
      alt=""
      src="http://static.boredpanda.com/blog/wp-content/uploads/2015/09/cute-bunnies-110__605.jpg"
      style={
        Object {
          "height": "100px",
          "width": "100px",
        }
      }
    />
    <figcaption>
       
      Cutest Bunny
       
    </figcaption>
  </figure>
</div>
`;